include/os/freebsd/spl/sys/cred.h

```diff
@@ -93,7 +93,7 @@ extern cred_t *zone_kcred(void);
 extern gid_t crgetrgid(const cred_t *);
 extern gid_t crgetsgid(const cred_t *);
 
-#define	crgetzoneid(x) (0)
+#define	crgetzoneid(cr) ((cr)->cr_prison)
```

Comment:
While this and below defines matching each other should work, I can't say that I very much like the type change from original Illumos int/long to pointer here.  Can we instead do (cr)->cr_prison->pr_id here and leave GLOBAL_ZONEID 0?  It means one more pointer dereference, but the only use of this function does not look like in particularly busy path.